distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Widget_Helpers.enso

```diff
@@ -117,7 +117,8 @@ make_column_ref_by_name_selector_for_type table display:Display=..Always value_t
     expression = if table.is_nothing then [] else [Option "<Expression>" "(expr '["+table.column_names.first+"]')"]
 
     col_names = if table.is_nothing then [] else
-        names = if is_all then table.column_names else (table.columns_helper.select_columns [..By_Type value_type, ..By_Type Value_Type.Mixed] case_sensitivity=..Default reorder=False error_on_missing_columns=False on_problems=..Ignore error_on_empty=False . map .name)
+        type_selector = if value_type.is_numeric && if_value_type_numeric_include_all_numeric then Value_Type.numeric_types.map t->(..By_Type t) else [..By_Type value_type]       
+        names = if is_all then table.column_names else (table.columns_helper.select_columns (type_selector + [..By_Type ..Mixed]) case_sensitivity=..Default reorder=False error_on_missing_columns=False on_problems=..Ignore error_on_empty=False . map .name)
```

Comment:
We are doing lots of layers here - in `numeric_types` we select types that `is_numeric = True` and then do a series of select by type on it. Wouldn't it be easier to just have a special case for numeric?

Although just a loose suggestion, maybe current one is better, I guess it's a bit subjective.

```suggestion
        names = if is_all then table.column_names else 
            if value_type.is_numeric && if_value_type_numeric_include_all_numeric then table.columns.filter (c-> c.value_type.is_numeric || c.value_type.is_mixed) . map .name else
                table.columns_helper.select_columns ([..By_Type value_type, ..By_Type ..Mixed]) case_sensitivity=..Default reorder=False error_on_missing_columns=False on_problems=..Ignore error_on_empty=False . map .name
```

Reply:
Yes I like this